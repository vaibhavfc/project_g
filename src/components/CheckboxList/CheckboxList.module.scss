@import '../../sass/theme/default';

.root {
  @include body-2;
  color: $checkbox-square;
  margin-bottom: -0.625rem;

  &:hover {
    background-color: rgba($checkbox-hover-unchecked, 0.08);
  }

  &:focus {
    background-color: rgba($checkbox-hover-unchecked, 0.08);
    transition: none;
  }
}

.checked {
  color: $checkbox-checked;

  &:hover {
    background-color: rgba($checkbox-hover-checked, 0.08);
  }

  &:focus {
    background-color: rgba($checkbox-hover-checked, 0.08);
    transition: none;
  }
}

.disabledChecked {
  color: $checkbox-disabled;
}

.label {
  @include body-2;
  transform: translate(-0.375rem, 0.3125rem);
  color: $checkbox-label;
}

.formLabel {
  @include body-3;
  color: $checkbox-label !important;
  text-transform: capitalize;
}
