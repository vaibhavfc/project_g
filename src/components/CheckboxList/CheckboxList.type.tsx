// import { Ref } from 'react';
import { CheckboxProps as MuiCheckboxProps, CheckboxClassKey } from '@material-ui/core/Checkbox';

export interface CheckboxListStylingProps extends Partial<Record<CheckboxClassKey, string>> {
}

export interface CheckboxListProps extends MuiCheckboxProps {
  // innerRef: Ref<HTMLButtonElement>;
  list: { title: string,
          items: {
            label: string,
            checked?: boolean,
            disabled?: boolean
          }[]
        };
}
