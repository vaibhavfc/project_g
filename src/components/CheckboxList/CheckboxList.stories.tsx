// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import CheckboxList from '.';

import { CheckboxListProps } from './CheckboxList.type';

export default {
  title: 'Components/CheckboxList',
  component: CheckboxList,
  parameters: {
    controls: {
      // disabled: true,
    }
  }
} as Meta;

const Template: Story<CheckboxListProps> = (args) => <CheckboxList {...args} />

export const Default = Template.bind({});
Default.args = {
  list: {
    title: 'Checkbox List Heading',
    items: [
      {label: 'Default'},
      {label: 'Checked', checked: true},
      {label: 'Disabled', disabled: true},
      {label: 'Checked & Disabled', disabled: true, checked: true}
    ]
  }
}
