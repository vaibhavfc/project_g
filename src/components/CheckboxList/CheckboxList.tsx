import React, { FC, useMemo } from 'react';
import {
  Checkbox as MuiCheckbox,
  FormControl,
  FormControlLabel,
  FormGroup,
  FormLabel,
} from '@material-ui/core';
import classnames from 'classnames';
import { CheckboxListProps, CheckboxListStylingProps } from './CheckboxList.type';

import classes from './CheckboxList.module.scss';

import { mergeClassesObjects } from '../../helpers/styling/mergeClassesObjects';

const CheckboxList: FC<CheckboxListProps> = ({
  // innerRef = null,
  classes: overrideClasses = {},
  // children = 'CheckboxList',
  tabIndex = 0,
  list,
  // ...props
}) => {
  const mergedClasses = useMemo(
    () => mergeClassesObjects<CheckboxListStylingProps>(classes, overrideClasses),
    [overrideClasses],
  );

  // const [clicked, setClicked] = React.useState(false);

  // const handleChange = (event: React.ChangeEvent<HTMLButtonElement>) => {
  //   setClicked(event.target.checked);
  //   return clicked;
  // };

  return (
    <FormControl component="fieldset" className={classes.formControl}>
      <FormLabel
        component="legend"
        classes={{
          root: classnames(classes.formLabel),
          // focused: classnames(classes.formLabel)
        }}
      >
        {list.title}
      </FormLabel>
      <FormGroup>
        {
          list.items?.map((checkbox) => (
            <FormControlLabel
              classes={{
                label: classes.label,
              }}
              value="end"
              control={(
                <MuiCheckbox
                  disabled={checkbox.disabled}
                  checked={checkbox.checked}
                  tabIndex={tabIndex}
                  classes={{
                    root: mergedClasses.root,
                    checked: classnames({
                      [classes.disabledChecked]: (checkbox.disabled && checkbox.checked),
                      [classes.checked]: (checkbox.checked && !checkbox.disabled),
                    }),
                  }}
                />
                )}
              label={checkbox.label}
              labelPlacement="end"
              key={checkbox.label}
            />
          ))
        }
      </FormGroup>
    </FormControl>
  );
};

// const CheckboxList = forwardRef<HTMLInputElement, Omit<CheckboxListProps, 'innerRef'>>(
//   (props, ref) => <CheckboxListWithInnerRef innerRef={ref} {...props} />
// );

export default CheckboxList;
