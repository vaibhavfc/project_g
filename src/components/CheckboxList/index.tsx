import CheckboxList from './CheckboxList';
import { CheckboxListProps } from './CheckboxList.type';

export type { CheckboxListProps };

export default CheckboxList;
