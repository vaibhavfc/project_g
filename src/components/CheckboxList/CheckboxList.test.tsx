import { render } from '@testing-library/react';

import renderer from 'react-test-renderer';

import CheckboxList from './CheckboxList';

const list = {
  title: 'Checkbox List Heading',
  items: [
    { label: 'Default' },
    { label: 'Checked', checked: true },
    { label: 'Disabled', disabled: true },
    { label: 'Checked & Disabled', disabled: true, checked: true },
  ],
};
describe('test', () => {
  it('Renders CheckboxList Compoents', () => {
    render(<CheckboxList list={list} />);
  });

  it('Matches Snapshot of Checklist', () => {
    const tree = renderer.create(<CheckboxList list={list} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
