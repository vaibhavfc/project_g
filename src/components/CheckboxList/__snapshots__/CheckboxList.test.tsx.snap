// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Matches Snapshot of Checklist 1`] = `
<fieldset
  className="MuiFormControl-root formControl"
>
  <legend
    className="MuiFormLabel-root formLabel"
  >
    Checkbox List Heading
  </legend>
  <div
    className="MuiFormGroup-root"
  >
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            className="PrivateSwitchBase-input-8"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            tabIndex={0}
            type="checkbox"
            value="end"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label label MuiTypography-body1"
      >
        Default
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-6 Mui-checked checked MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input-8"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            tabIndex={0}
            type="checkbox"
            value="end"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label label MuiTypography-body1"
      >
        Checked
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root Mui-disabled"
    >
      <span
        aria-disabled={true}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-disabled-7 Mui-disabled MuiIconButton-colorSecondary Mui-disabled Mui-disabled"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            className="PrivateSwitchBase-input-8"
            data-indeterminate={false}
            disabled={true}
            onChange={[Function]}
            tabIndex={0}
            type="checkbox"
            value="end"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label label Mui-disabled MuiTypography-body1"
      >
        Disabled
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root Mui-disabled"
    >
      <span
        aria-disabled={true}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-6 Mui-checked disabledChecked PrivateSwitchBase-disabled-7 Mui-disabled MuiIconButton-colorSecondary Mui-disabled Mui-disabled"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input-8"
            data-indeterminate={false}
            disabled={true}
            onChange={[Function]}
            tabIndex={0}
            type="checkbox"
            value="end"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label label Mui-disabled MuiTypography-body1"
      >
        Checked & Disabled
      </span>
    </label>
  </div>
</fieldset>
`;
